### PR TITLE
Add script to fetch Uniprot data

### PR DESCRIPTION
**Pull request by seeta-ramaraju**
[G2P-33](https://www.ebi.ac.uk/panda/jira/browse/G2P-33)

### CODE REVIEW

File: uniprot/uniprot_importer.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import requests
+from requests.adapters import HTTPAdapter, Retry
+import re
+import MySQLdb
+import datetime
+import argparse
+
+# Uniprot data fetch URL
+url = 'https://rest.uniprot.org/uniprotkb/search?query=accession:P83916&fields=accession,cc_function,xref_mim,xref_hgnc,gene_primary' # To be changed
```
Comment (dglemos) [inline uniprot/uniprot_importer.py:9]: The script is only importing this gene.

File: uniprot/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Uniprot Import Script
```
Comment (dglemos) [inline uniprot/README.md:1]: The README file content could be moved to the beginning of the `uniprot_importer.py`
This way the info to run the script is in the same file as the script.

**Review by dglemos — COMMENTED** [review]
In the future we are going to have other types of scripts and pipelines in this repo.
To prepare for that, you could create the following structure:

gene2phenotype_scripts/
-- scripts/
         --import/
                  -- uniprot/
                           -- uniprot_importer.py

File: uniprot/uniprot_importer.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    if len(data) != 0:
+        for row in data:
+            identifier_to_locus_id_map[row[0]] = row[1]
+    # Insert Uniprot data
+    insert_count = 0
+    for item in total_items:
+        if item["HGNC"] in identifier_to_locus_id_map:
+            cursor.execute(insert_sql, [item["accession"],
+                                  identifier_to_locus_id_map[item["HGNC"]],
+                                  int(item["HGNC"][5:]), # To be changed once column datatype changes
```
Comment (dglemos) [inline uniprot/uniprot_importer.py:101]: You could update the include the prefix "HGNC". We are only going to run the import when the db schema is updated to varchar.

Comment (seeta-ramaraju) [reply uniprot/uniprot_importer.py]: taken care

**Comment by seeta-ramaraju** [comment]
> In the future we are going to have other types of scripts and pipelines in this repo. To prepare for that, you could create the following structure:
> 
> ```
> gene2phenotype_scripts/
> -- scripts/
>          --import/
>                   -- uniprot/
>                            -- uniprot_importer.py
> ```

taken care in latest commit

File: scripts/import/uniprot/uniprot_importer.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+def is_protein_function_available(dataItem):
+    return "comments" in dataItem and len(dataItem["comments"])!=0 and "texts" in dataItem["comments"][0] and len(dataItem["comments"][0]["texts"])!=0 and "value" in dataItem["comments"][0]["texts"][0];
+
+# Function to fetch Uniprot data
+def fetch_all_data():
+    total_items = []
+    for batch in get_batch(url):
+        current_batch_json = batch.json()
+        for item in current_batch_json["results"]:
+            # If protein function or HGNC id is not available then dont consider the data entry
```
Comment (dglemos) [inline scripts/import/uniprot/uniprot_importer.py:65]: dont -> don't

**Review by dglemos — COMMENTED** [review]
The new path looks good, thanks!

There are still two comments you need to address:
- https://github.com/EBI-G2P/gene2phenotype_scripts/pull/1#discussion_r1557233932
- https://github.com/EBI-G2P/gene2phenotype_scripts/pull/1#discussion_r1557429164

File: scripts/import/uniprot/uniprot_importer.py (hunk 4 above)
Comment (seeta-ramaraju) [reply scripts/import/uniprot/uniprot_importer.py]: taken care in latest commit

File: uniprot/uniprot_importer.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+def get_database_value(database, dataItem):
+    if "uniProtKBCrossReferences" in dataItem and len(dataItem["uniProtKBCrossReferences"]):
+        for item in dataItem["uniProtKBCrossReferences"]:
+            if item["database"] == database:
+                return item["id"]
+    return None
+
+def is_protein_function_available(dataItem):
+    return "comments" in dataItem and len(dataItem["comments"])!=0 and "texts" in dataItem["comments"][0] and len(dataItem["comments"][0]["texts"])!=0 and "value" in dataItem["comments"][0]["texts"][0];
```
[the remark this answers is not in the document]
Comment (seeta-ramaraju) [reply uniprot/uniprot_importer.py]: taken care in latest commit

**Pull request merged**